### ISSUE TITLE
syntax highlighting for native pipe `|>`

### ISSUE THREAD

**Issue by stragu** [tidyverse/reprex#389]
Currently, using the default syntax highlighting, the native pipe is highlighted as separate `|` and `>`. The following:

``` r
library(magrittr)
1.23 %>% round()
#> [1] 1
1.23 |> round()
#> [1] 1
```

<sup>Created on 2021-08-16 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Results in this in the viewer:

![image](https://user-images.githubusercontent.com/1747497/129503720-0d38f42d-0266-4bca-a148-760fcc7412b3.png)

It would be great to support highlighting the native pipe as a distinct operator.

**Comment by cderv** [comment]
I believe this should be solved in **rmarkdown** directly. We already released (in **rmarkdown** 2.10) a quick fix for `html_document()` related output, but I don't think **reprex** is using `html_document()` or `html_document_base()` to render the HTML preview. It is the same for `github_document()` - the highlighting oddity still remains.

We plan to make a proper fix that **reprex** could benefit. We are tracking this in https://github.com/rstudio/rmarkdown/issues/2196

**Comment by jennybc** [comment]
Looks like reprex just needs to wait for the upstream fixes to happen, as described above. It's really not something the reprex package has control over.
